packages/mojatax/src/Utils.js

```diff
@@ -0,0 +1,87 @@
+import { composeNextState } from '@openfn/language-common';
+import {
+  request as commonRequest,
+  assertRelativeUrl,
+  logResponse,
+} from '@openfn/language-common/util';
+import nodepath from 'node:path';
+
+export const authorize = state => {
+  const auth = state.configuration;
+  if (auth.access_token) {
+    return state;
+  }
+  const clientId = auth.clientId;
+  const password = auth.password;
+  const headers = {};
+
+  if (clientId && password) {
+    Object.assign(headers, {
+      'Content-Type': 'application/json',
+    });
+
+    const body = {
+      client_id: clientId,
+      password: password,
+    };
+    const options = {
+      body: JSON.stringify(body),
+      headers,
+      method: 'POST',
+      parseAs: 'json',
+      baseUrl: auth.baseUrl,
+    };
+
+    return commonRequest('POST', '/api/v1/auth/clientLogin', options).then(
+      response => {
+        return {
+          ...state,
+          configuration: {
+            ...state.configuration,
+            access_token: response.body.access_token,
+          },
+        };
+      }
+    );
+  } else {
+    throw new Error(
+      'Invalid authorization credentials. Include clientId and password in state.configuration'
+    );
+  }
+};
+
+export const prepareNextState = (state, response, callback = s => s) => {
+  const { body, ...responseWithoutBody } = response;
+
+  const nextState = {
+    ...composeNextState(state, response.body),
+    response: responseWithoutBody,
+  };
+
+  return callback(nextState);
+};
+
+export const request = (configuration, method, path, options) => {
+  const { baseUrl, access_token } = configuration;
```

Comment:
Again, and I'm being paranoid really because this came up recently, can we just ensure configuration is defined:
```
const { baseUrl, access_token } = configuration ?? {};
```